libtenzir/builtins/operators/deduplicate.cpp

```diff
@@ -187,8 +187,9 @@ class deduplicate_operator final : public crtp_operator<deduplicate_operator> {
 
   auto optimize(expression const& filter, event_order order) const
     -> optimize_result override {
+    (void)filter;
     (void)order;
-    return optimize_result{filter, event_order::schema, copy()};
+    return do_not_optimize(*this);
```

Comment:
I think you could conditionally propagate the filter, if there is no `--limit` set, right? But `event_order::schema` looks incorrect in either case.

Reply:
Right, that's a good call. I didn't actually notice that this is only broken because of `--limit`.

Reply:
Eh, I (and probably you) mean `--distance`, not `--limit`, right?

Reply:
Yes. 🙈